lib/wallet/src/Cardano/Wallet/Primitive/Types/Credentials.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Cardano.Wallet.Primitive.Types.Credentials
+    ( Credentials (..)
+    , PrivateKey
+    , HashedCredentials
+    , ClearCredentials
+    ) where
+
+import Prelude
+
+
+import Cardano.Address.Derivation
+    ( XPrv )
+import Cardano.Wallet.Address.Derivation
+    ( Depth (RootK) )
+import Cardano.Wallet.Primitive.Passphrase.Types
+    ( Passphrase, PassphraseHash )
+
+type PrivateKey k = k 'RootK XPrv
+
+-- | A 'PrivateKey' for a given 'KeyFlavor'.
+data Credentials k pw = Credentials
```

Comment:
```suggestion
data RootCredentials k pw = RootCredentials
```

🤔 As the term "credentials" is also overloaded (though not as heavily as PrivateKey), I think it's a good idea to add the word "root" as well — to make it clear that these credentials refer to the root private key.